framework/src/modules/dpos_v2/commands/claim_rewards.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright © 2021 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { CommandExecuteContext } from '../../../state_machine';
+import { BaseCommand } from '../../base_command';
+import { InternalMethod } from '../internal_method';
+import { DelegateStore } from '../stores/delegate';
+import { VoterStore } from '../stores/voter';
+
+export class ClaimRewardsCommand extends BaseCommand {
+	private _internalMethod!: InternalMethod;
+
+	public addDependencies(args: { internalMethod: InternalMethod }) {
+		this._internalMethod = args.internalMethod;
+	}
+
+	public async execute(context: CommandExecuteContext<Record<string, never>>): Promise<void> {
+		const {
+			transaction: { senderAddress },
+			getMethodContext,
+		} = context;
+
+		const delegateStore = this.stores.get(DelegateStore);
+		const delegate = await delegateStore.get(context, senderAddress);
+		const voterStore = this.stores.get(VoterStore);
+		const voterData = await voterStore.get(context, senderAddress);
+		for (const sentVote of voterData.sentVotes) {
+			await this._internalMethod.assignVoteRewards(
```

Comment:
delegate should be obtained per sentVote, and voter data needs to be updated with the delegate sharing coefficient like [here](https://github.com/LiskHQ/lisk-sdk/commit/59a84fa7deab822291d5bee4af0cbba30a93a772#diff-5fd3e0c3e3b0fbac86b4bf6f8819ec635ad9c286218c4b09075232ba35bfdb44R221-R230)

Reply:
Shouldn't this be mentioned in the specification ?
```
for i in range(len(voterStore(senderAddress).sentVotes)):
        assignVoteRewards(senderAddress, i)
```

Reply:
in the LIP, it's getting the delegate in the loop `for elem in delegateStore(vote.delegateAddress).sharingCoefficients:`, but it's updated because it was inefficient in the vote scenario